index.js

```diff
@@ -41,23 +41,31 @@ module.exports = function pluginsrc (ctx, path, options) {
 
   if (!ctx.env) process.env.NODE_ENV = 'development'
 
-  var file
+  function handleResult (result) {
+    if (!result) throw new Error('No PostCSS Config found in: ' + path)
 
-  return config('postcss', options)
-    .load(path)
-    .then(function (result) {
-      if (!result) throw new Error('No PostCSS Config found in: ' + path)
+    file = result ? result.filepath : ''
 
-      file = result ? result.filepath : ''
+    return result ? result.config : {}
+  }
 
-      return result ? result.config : {}
-    })
-    .then(function (plugins) {
-      if (typeof plugins === 'function') plugins = plugins(ctx)
-      else plugins = assign(plugins, ctx)
+  function composePlugins (plugins) {
+    if (typeof plugins === 'function') plugins = plugins(ctx)
+    else plugins = assign(plugins, ctx)
 
-      if (!plugins.plugins) plugins.plugins = []
+    if (!plugins.plugins) plugins.plugins = []
 
-      return { plugins: loadPlugins(plugins), file: file }
-    })
+    return { plugins: loadPlugins(plugins), file: file }
+  }
+
+  var file
+
+  if (options.sync) {
+    return composePlugins(handleResult(config('postcss', options).load(path)))
```

Comment:
config is still asycn i tried running this code should we use sync api instead?
https://github.com/davidtheclark/cosmiconfig#synchronous-api